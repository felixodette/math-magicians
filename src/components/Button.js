import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Button extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const {buttonClass, eventHandler, value} = this.props;
    return (
      <button type="button" className={buttonClass} onClick={eventHandler}>{value}</button>
    );
  }
}

Button.propTypes = {
  buttonClass: PropTypes.string,
  eventHandler: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};


