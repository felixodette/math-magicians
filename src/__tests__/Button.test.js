import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import Button from '../components/Button';
import '@testing-library/jest-dom';

describe('Button componenet testing', () => {
  const RESULT = jest.fn();

  it('Rendered', () => {
    const BUTTON = renderer
      .create(<Button eventHandler={RESULT} value="Test" />)
      .toJSON();
    expect(BUTTON).toMatchSnapshot();
  });

  it('Click', () => {
    render(<Button eventHandler={RESULT} value="Test" />);
    fireEvent.click(screen.getByText('Test'));
    fireEvent.click(screen.getByText('Test'));
    fireEvent.click(screen.getByText('Test'));
    fireEvent.click(screen.getByText('Test'));
    expect(RESULT).toHaveBeenCalledTimes(4);
  });
});
