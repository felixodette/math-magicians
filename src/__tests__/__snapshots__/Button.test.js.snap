// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button componenet testing Rendered 1`] = `
<button
  className=""
  onClick={[MockFunction]}
  type="button"
>
  Test
</button>
`;
